src/Draco.Compiler/Api/Diagnostics/Diagnostic.cs

```diff
@@ -37,9 +37,29 @@ public sealed record class DiagnosticTemplate
     /// <param name="title">A short title for the message.</param>
     /// <param name="severity">The severity of the message.</param>
     /// <param name="format">The format string that describes the diagnostic in detail.</param>
+    /// <param name="errorCode">The error code of the error.</param>
     /// <returns>The constructed <see cref="DiagnosticTemplate"/>.</returns>
-    public static DiagnosticTemplate Create(string title, DiagnosticSeverity severity, string format) =>
-        new(title: title, severity: severity, format: format);
+    public static DiagnosticTemplate Create(string title, DiagnosticSeverity severity, string format, string errorCode) =>
+        new(title: title, severity: severity, format: format, errorCode: errorCode);
+
+    // NOTE: error codes are using format DRX###
+    // where X is category of the error in form of number and ### is index of the error,
+    // every category starts indexing from 1 except of internal compiler error
+    // Categories:
+    // 0 - Internal compiler error
+    // 1 - Syntax error
+    // 2 - Symbol resolution error
+    // 3 - Type checking error
+    // 4 - Dataflow error
+    // 5 - Codegen error
+
+    /// <summary>
+    /// Creates error code for draco error.
+    /// </summary>
+    /// <param name="category">Category of the error.</param>
+    /// <param name="index">Index of the error.</param>
+    /// <returns></returns>
+    public static string SyntaxErrorCode(int category, int index) => $"DR{category}{index:D3}";
```

Comment:
Name is misleading, then it's `ErrorCode`. A `SyntaxErrorCode` would return an error code with the syntax errors category